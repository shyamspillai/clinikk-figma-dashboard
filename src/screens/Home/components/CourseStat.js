import styled from 'styled-components'

const CourseStat = ({statNumber, statType, statStatus}) => {
    return (
        <StatComponent>
            <span className="baloo size-60 weight-600 font-black">{statNumber}</span>
            <div className="flex column">
                <span className="baloo size-16 weight-400 font-black">{statType}</span>
                <span className="baloo size-16 weight-400 font-black">{statStatus}</span>
            </div>
        </StatComponent>
    )
}

const StatComponent = styled.div`
    flex: 0.4;
    display: flex;
    flex-direction: row;
    background-color: #f5f5f7;
    justify-content: space-around;
    align-items: center;
    padding: 15px 15px;
    border-radius: 15px;

    @media only screen and (max-width: 400px) {
        flex: 0.45
    }
`;

export default CourseStat
