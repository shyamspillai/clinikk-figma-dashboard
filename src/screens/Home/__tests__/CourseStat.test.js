import React from 'react';
import CourseStat from '../components/CourseStat';

import Enzyme, { render, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe('Course Stat component', () => {

  const props = {
    statNumber: 5,
    statType: 'course',
    statStatus: 'in progress'
  }

  it('should match snapshot', () => {
    const CourseStatComponent = shallow(<CourseStat />);
    expect(toJson(CourseStatComponent)).toMatchSnapshot();
  })

  it('should render the course stat props correctly', () => {
    const wrapper = render(
      <CourseStat {...props} />
    );

    const numberSpan = wrapper.find('span.size-60').first()
    const typeSpan = wrapper.find('span.size-16').first()
    const statusSpan = wrapper.find('span.size-16').last()

    expect(numberSpan.text()).toEqual('5')
    expect(typeSpan.text()).toEqual('course')
    expect(statusSpan.text()).toEqual('in progress')
  });

  it('should render the course stat props correctly in mobile view', () => {
    const wrapper = render(
      <CourseStat {...props} />
    );

    // Change the viewport to 500px.
    global.innerWidth = 500;

    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize'));

    const numberSpan = wrapper.find('span.size-60').first()
    const typeSpan = wrapper.find('span.size-16').first()
    const statusSpan = wrapper.find('span.size-16').last()

    expect(numberSpan.text()).toEqual('5')
    expect(typeSpan.text()).toEqual('course')
    expect(statusSpan.text()).toEqual('in progress')
  });

});