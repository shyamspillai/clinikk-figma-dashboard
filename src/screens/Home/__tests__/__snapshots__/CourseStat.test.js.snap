// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Stat component should match snapshot 1`] = `
<styled.div>
  <span
    className="baloo size-60 weight-600 font-black"
  />
  <div
    className="flex column"
  >
    <span
      className="baloo size-16 weight-400 font-black"
    />
    <span
      className="baloo size-16 weight-400 font-black"
    />
  </div>
</styled.div>
`;
